org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/proposals/JavadocTagsSubProcessor.java

```diff
@@ -14,440 +14,139 @@
-		proposals.add(CodeActionHandler.wrap(proposal, CodeActionKind.QuickFix));
-
-		String label2= CorrectionMessages.JavadocTagsSubProcessor_addjavadoc_allmissing_description;
-		ASTRewriteCorrectionProposalCore addAllMissing = new AddAllMissingJavadocTagsProposalCore(label2, context.getCompilationUnit(), bodyDeclaration, IProposalRelevance.ADD_ALL_MISSING_TAGS);
-		proposals.add(CodeActionHandler.wrap(addAllMissing, CodeActionKind.QuickFix));
+		new JavadocTagsSubProcessor().addMissingJavadocTagProposals(context, problem, proposals);
 	}
 
 	public static void getUnusedAndUndocumentedParameterOrExceptionProposals(IInvocationContextCore context,
 			IProblemLocationCore problem, Collection<ProposalKindWrapper> proposals) {
-		ICompilationUnit cu= context.getCompilationUnit();
-		IJavaProject project= cu.getJavaProject();
-
-		if (!JavaCore.ENABLED.equals(project.getOption(JavaCore.COMPILER_DOC_COMMENT_SUPPORT, true))) {
-			return;
-		}
-
-		int problemId= problem.getProblemId();
-		boolean isUnusedTypeParam= problemId == IProblem.UnusedTypeParameter;
-		boolean isUnusedParam= problemId == IProblem.ArgumentIsNeverUsed || isUnusedTypeParam;
-		String key= isUnusedParam ? JavaCore.COMPILER_PB_UNUSED_PARAMETER_INCLUDE_DOC_COMMENT_REFERENCE : JavaCore.COMPILER_PB_UNUSED_DECLARED_THROWN_EXCEPTION_INCLUDE_DOC_COMMENT_REFERENCE;
-
-		if (!JavaCore.ENABLED.equals(project.getOption(key, true))) {
-			return;
-		}
-
-		ASTNode node= problem.getCoveringNode(context.getASTRoot());
-		if (node == null) {
-			return;
-		}
-
-		BodyDeclaration bodyDecl= ASTResolving.findParentBodyDeclaration(node);
-		if (bodyDecl == null || ASTResolving.getParentMethodOrTypeBinding(bodyDecl) == null) {
-			return;
-		}
-
-		String label;
-		if (isUnusedTypeParam) {
-			label= CorrectionMessages.JavadocTagsSubProcessor_document_type_parameter_description;
-		} else if (isUnusedParam) {
-			label= CorrectionMessages.JavadocTagsSubProcessor_document_parameter_description;
-		} else {
-			node= ASTNodes.getNormalizedNode(node);
-			label= CorrectionMessages.JavadocTagsSubProcessor_document_exception_description;
-		}
-		ASTRewriteCorrectionProposalCore proposal = new AddMissingJavadocTagProposalCore(label, context.getCompilationUnit(), bodyDecl, node, IProposalRelevance.DOCUMENT_UNUSED_ITEM);
-		proposals.add(CodeActionHandler.wrap(proposal, CodeActionKind.QuickFix));
+		new JavadocTagsSubProcessor().addUnusedAndUndocumentedParameterOrExceptionProposals(context, problem, proposals);
 	}
 
-	public static void getMissingJavadocCommentProposals(IInvocationContextCore context, ASTNode node,
+	public static void getMissingJavadocCommentProposals(IInvocationContextCore context, ASTNode coveringNode,
 			Collection<ProposalKindWrapper> proposals, String kind) throws CoreException {
-		if (node == null) {
-			return;
-		}
-		BodyDeclaration declaration= ASTResolving.findParentBodyDeclaration(node);
-		if (declaration == null || declaration.getJavadoc() != null) {
-			return;
-		}
```

Comment:
In the previous behaviour, this ensured we never suggested adding new javadoc where one already existed. With the new behaviour we always suggest it. Seems wrong.

It looks like this is missing upstream.

Reply:
I've just added an override of the base class implementation to add this check.